libass/ass_fontselect.c

```diff
@@ -542,17 +542,15 @@ find_font(ASS_FontSelector *priv, ASS_Library *library,
             ASS_FontInfo *font = &priv->font_infos[x];
             unsigned score = UINT_MAX;
 
-            if (matches_family_name(font, fullname)) {
-                // If there's a family match, compare font attributes
-                // to determine best match in that particular family
-                score = font_attributes_similarity(font, &req);
-                *name_match = true;
-            } else if (matches_full_or_postscript_name(font, fullname)) {
```

Comment:
I suppose this would stop being flat-out wrong if you just reverted this logic reordering.

But this would still be no more than a bandaid. But maybe it’s a good bandaid.